plotly/tools.py

```diff
@@ -1269,3 +1277,628 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
             )
 
     return figure
+
+
+class TraceFactory(dict):
+    @staticmethod
+    def create_quiver(x, y, u, v, scale=.1, arrow_scale=.3,
+                      angle=math.pi/9, **kwargs):
+        """
+        Return data for a quiver plot.
+
+        :param (list or array) x: x coordinates of the arrow locations
```

Comment:
:cow2: the stuff in the `()` in should be real types. do you mean `(list or ndarray)`? This can also be written `(list|ndarray)`. Docstring parsers can then give you type hinting :)